src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +151,1301 @@ static DamageTable D_80B65044 = {
+        if ((this->actor.isTargeted) && (this->actionFunc != EnBigpo_WarpingOut) &&
+            !(this->collider.base.acFlags & AC_HARD) && (this->actor.category == ACTORCAT_ENEMY)) {
+            this->unk20C++;
+        } else {
+            this->unk20C = 0;
+        }
+        if (this->unk20C == 40) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_LAUGH);
+            EnBigpo_SetupWarpOut(this);
+        }
+    }
+
+    this->actionFunc(this, globalCtx);
+    if ((this->actionFunc != EnBigpo_SpawnPoCutscene6) && (this->actionFunc != EnBigpo_SpawnPoCutscene4)) {
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
+    if (this->actionFunc == EnBigpo_LanternFalling) {
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 27.0f, 60.0f, 4);
+    }
+
+    if (this->actor.draw == EnBigpo_DrawScoopSoul) {
+        Actor_SetHeight(&this->actor, -10.0f);
+    } else {
+        Actor_SetHeight(&this->actor, 42.0f);
+    }
+
+    EnBigpo_UpdateColor(this);
+    EnBigpo_FlickerLanternLight(this);
+
+    this->actor.shape.shadowAlpha = this->mainColor.a;
+    thisCollider = &this->collider;
+    Collider_UpdateCylinder(&this->actor, thisCollider);
+    if (this->collider.base.ocFlags1 & OC1_ON) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &thisCollider->base);
+    }
+    if (this->collider.base.atFlags & AT_ON) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &thisCollider->base);
+    }
+    if (this->collider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &thisCollider->base);
+    }
+
+    if (this->unk21C > 0.0f) {
+        Math_StepToF(&this->unk21C, 0.0f, 0.05f);
+        if (this->mainColor.a != 255) { // NOT fully visible
+            if (this->mainColor.a * (1.0f / 255.0f) < this->mainColor.a) {
+                this->unk21C = this->mainColor.a * (1.0f / 255.0f);
+            }
+        }
+        this->unk220 = ((this->unk21C + 1.0f) * 0.5f);
+        this->unk220 = CLAMP_MAX(this->unk220, 1.0f);
+    }
+}
+
+// alt update func: the revealed fires under dampe's house
+void EnBigpo_UpdateFire(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    // bilboarding the fire toward player camera?
+    this->actor.shape.rot.y = BINANG_ROT180(func_800DFCDC(ACTIVE_CAM));
+    this->actionFunc(this, globalCtx);
+}
+
+s32 EnBigpo_OverrideLimbDraw2(struct GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                              struct Actor* actor, Gfx** gfx) {
+    EnBigpo* this = (EnBigpo*)actor;
+    // not fully invisible
+    if ((!((this->mainColor.a != 0) && (limbIndex != 7))) ||
+        ((this->actionFunc == EnBigpo_BurnAwayDeath) && (this->idleTimer >= 2))) {
+        *dList = NULL;
+    }
+    return 0;
+}
+
+void EnBigpo_PostLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor,
```

Comment:
What is the difference between PostLimbDraw and PostLimbDraw2

Reply:
This function gets passed to SkelAnime_Draw2 which takes a OverrideLimbDraw2 and PostLimbDraw2 type function pointers

https://github.com/zeldaret/mm/blob/master/include/z64animation.h#L187-L188

I left the 2 part of renaming these functions to `EnBigpo_OverrideLimbDraw2` and `EnBigpo_PostLimbDraw2`. Should the 2 be removed because the type is not as important here as people might try to read the code and think they are the second of their functions?

Reply:
Yeah I would say remove the 2